lapis/src/test/kotlin/org/genspectrum/lapis/model/AdvancedQueryFacadeTest.kt

```diff
@@ -0,0 +1,774 @@
+                        IntBetween("intField", 1, null),
+                    ),
+                ),
+                ValidTestCase(
+                    "floatField",
+                    "floatField>=0 AND floatField<=10e-1",
+                    And(
+                        FloatBetween("floatField", null, 1.0),
+                        FloatBetween("floatField", 0.0, null),
+                    ),
+                ),
+                ValidTestCase(
+                    "date",
+                    "date>=2020-01-01 AND date<=2021-01-01",
+                    And(
+                        DateBetween("date", null, LocalDate.parse("2021-01-01")),
+                        DateBetween("date", LocalDate.parse("2020-01-01"), null),
+                    ),
+                ),
+            )
+
+        @JvmStatic
+        fun invalidQueryProvider() =
+            listOf(
+                InvalidTestCase(
+                    "date >= and <=",
+                    "date<=2021.01.01",
+                    "'2021.01.01' is not a valid date",
+                ),
+                InvalidTestCase(
+                    "metadata >= and <=",
+                    "some_metadata<=2021.01.01",
+                    "expression <= cannot be used for field some_metadata of type STRING",
+                ),
+                InvalidTestCase(
+                    "intField >= and <=",
+                    "intField>=One",
+                    "'One' is not a valid int",
+                ),
+                InvalidTestCase(
+                    "floatField >= and <=",
+                    "floatField>=One",
+                    "'One' is not a valid float",
+                ),
+                InvalidTestCase(
+                    "dateFrom",
+                    "dateFrom=2020-01-01",
+                    "Metadata field dateFrom does not exist",
+                ),
+                InvalidTestCase(
+                    "dateTo",
+                    "dateTo=2020-01-01",
+                    "Metadata field dateTo does not exist",
+                ),
+                InvalidTestCase(
+                    "intFieldFrom",
+                    "intFieldFrom=1",
+                    "Metadata field intFieldFrom does not exist",
+                ),
+                InvalidTestCase(
+                    "intFieldTo",
+                    "intFieldTo=1",
+                    "Metadata field intFieldTo does not exist",
+                ),
+                InvalidTestCase(
+                    "floatFieldFrom",
+                    "floatFieldFrom=1",
+                    "Metadata field floatFieldFrom does not exist",
+                ),
+                InvalidTestCase(
+                    "floatFieldTo",
+                    "floatFieldTo=1",
+                    "Metadata field floatFieldTo does not exist",
+                ),
+                InvalidTestCase(
+                    "date>= with IsNull",
+                    "IsNull(dateFrom>=2020-01-01)",
```

Comment:
We only have tests about what should not work in `isNull`, but we don't have tests about what should work.